src/sqlfluff/core/linter/linter.py

```diff
@@ -1103,17 +1103,39 @@ def lint_path(
         processes: Optional[int] = None,
     ) -> LintedDir:
         """Lint a path."""
-        linted_path = LintedDir(path)
-        if self.formatter:
-            self.formatter.dispatch_path(path)
-        fnames = list(
-            self.paths_from_path(
+        return self.lint_paths(
```

Comment:
I rewrote `lint_path()` to simply forward to `lint_paths()`, now that `lint_paths()` does all the work.

Reply:
Good idea for backward compatability